lua/harpoon/mark.lua

```diff
@@ -73,10 +92,7 @@ M.add_file = function(file_name_or_buf_id)
         return
     end
 
-    if buf_name == "" or buf_name == nil then
-        error("Couldn't find a valid file name to mark, sorry.")
-        return
-    end
+    validate_buf_name(buf_name)
```

Comment:
You dropped the early return

Reply:
Also above, it wont let me comment, but 5 lines above there is 

```
    if M.valid_index(M.get_index_of(buf_name)) then
        -- we don't alter file layout.
        return
    end
```

That should be replaced with this logic

```
if not validate_buf_name(buf_name) or mark_exists(buf_name) then
    return
end
```

Reply:
I didn't want to have to repeat this checking logic all over, that was the reason I made the validation function which stops execution if validation fails. As for the early return, it error raises so execution stops on its own.